File: androidHyperskillApp/src/main/java/org/hyperskill/app/android/sentry/domain/model/transaction/PlatformHyperskillSentryTransaction.kt
package org.hyperskill.app.android.sentry.domain.model.transaction

import io.sentry.ITransaction
import org.hyperskill.app.sentry.domain.model.transaction.HyperskillSentryTransaction

class PlatformHyperskillSentryTransaction(
    val transaction: ITransaction,
    tags: Map<String, String>
) : HyperskillSentryTransaction(transaction.name, transaction.operation, tags)

File: iosHyperskillApp/iosHyperskillApp/Sources/Systems/Sentry/PlatformHyperskillSentryTransaction.swift
import Foundation
import Sentry
import shared

final class PlatformHyperskillSentryTransaction: HyperskillSentryTransaction {
    let span: Span

    init(span: Span, name: String, operation: String, tags: [String: String]) {
        self.span = span
        super.init(name: name, operation: operation, tags_: tags)
    }
}
